zerver/lib/bulk_create.py

```diff
@@ -116,6 +118,26 @@ def bulk_create_users(
 
     Subscription.objects.bulk_create(subscriptions_to_create)
 
+    full_members_system_group = UserGroup.objects.get(
```

Comment:
This codepath is also called via `do_import_system_bots` calling `create_users`, which calls this `bulk_create_users` - in `import_realm.py` - have you checked if this will work fine? E.g. if importing from another server, which has these `UserGroupMembership`s in the internal realm, won't an error happen due to the `UserGroupMembership`s being attempted to be created twice - once through this codepath and once through the import of user groups?

I haven't looked at this in-depth, just an edge case that came to mind that is not clear to me if was addressed - we can audit further if needed

Reply:
I checked it once again now and this will work fine. `do_import_system_bots` is called after importing realm, and we also check whether the bots are already created or not and I think we can assume that the `UserGroupMemberships` objects in the import data will only be present if `UserProfile` objects are present in import data.

Reply:
Okay, cool thanks for checking!

Reply:
I think the comment here is incorrect right now, though, since this is called from the data import code path. However, as noted in the thread I started here:

https://chat.zulip.org/#narrow/stream/3-backend/topic/do_import_system_bots/near/1417790

I think that the data import code path is a noop (and would be a bug if it weren't) and should be deleted, so the comment is likely fine as written.